src/messages/textDocument_document.cc

```diff
@@ -165,88 +152,64 @@ void MessageHandler::textDocument_documentSymbol(JsonReader &reader,
     std::sort(result.begin(), result.end());
     reply(result);
   } else if (g_config->client.hierarchicalDocumentSymbolSupport) {
-    std::unordered_map<SymbolIdx, std::unique_ptr<DocumentSymbol>> sym2ds;
-    std::vector<std::pair<std::vector<const void *>, DocumentSymbol *>> funcs,
-        types;
+    std::vector<ExtentRef> syms;
+    syms.reserve(file->symbol2refcnt.size());
```

Comment:
Here I reserve probably too much space but at least there is only 1 allocation done. What do you think?

Reply:
This is fine. syms is about the same scale of symbol2refcnt.